index.js

```diff
@@ -47,7 +52,16 @@ datasets.forEach(dataset => {
   deleteEndpointDatasets.forEach(data => {
     app.delete(`${pathPrefix}/${data}`, (request, response) => {
       let newData = request.body;
-      app.locals.bookings = app.locals.bookings.filter(booking => booking.id !== newData.id);
+      app.locals[data] = app.locals[data].filter(el => el.id !== newData.id);
+      return response.sendStatus(200);
+    });
+  });
+
+  putEndpointDatasets.forEach(data => {
+    app.put(`${pathPrefix}/${data}`, (request, response) => {
+      let newData = request.body;
+      let index = app.locals[data].findIndex(el => el.id === newData.id);
+      app.locals[data][index] = newData;
```

Comment:
@hannahhch I think we may need some more error handling here. You can delete all of the information on an existing trip by just overwriting the data. I could see people messing things up on accident. Maybe make it so that they can only modify certain properties of the trip? Let's talk more about this / what the Agent's power should be in the app.

Reply:
Dangerous PUT:
<img width="734" alt="Screen Shot 2020-02-10 at 10 54 33 AM" src="https://user-images.githubusercontent.com/36432040/74175822-ca379d80-4bf3-11ea-915a-8b2e1140f996.png">

Resulting GET:
<img width="479" alt="Screen Shot 2020-02-10 at 10 54 41 AM" src="https://user-images.githubusercontent.com/36432040/74175832-ce63bb00-4bf3-11ea-95d4-2ba9285aa9fe.png">